feat(kno-7125): add relative delay documentation (#641)
knocklabs/docs: content/designing-workflows/delay-function.mdx

  </Accordion>
</AccordionGroup>

> ## Wait for a relative duration

Reviewer: ```suggestion
## Wait until a relative timestamp
```

My thinking here is language maps more closely to UI and to what this functionally doing. "Relative duration" kind of confuses me (could say it is same as "dynamic duration"?) but "wait until a relative timestamp" feels more specific to what this is doing.

> You can set a relative delay to compute a delay time from a set interval and a dynamic duration. This means you can set a delay that occurs a set amount of time before or after a provided dynamic timestamp. This allows the creation of flexible and context-aware delay periods in your workflows.

Reviewer 2: ```suggestion
You can set a relative delay to compute a delay time from a set interval and a dynamic timestamp. This means you can set a delay that occurs a set amount of time before or after a provided dynamic timestamp. This allows the creation of flexible and context-aware delay periods in your workflows.
```
Just realized we use it here too, but then later in the sentence say "timestamp." My personal preference is to just use "timestamp" consistently

> You can set a relative delay to compute a delay time from a set interval and a dynamic duration. This means you can set a delay that occurs a set amount of time before or after a provided dynamic timestamp. This allows the creation of flexible and context-aware delay periods in your workflows.

Reviewer: ```suggestion
You can use our relative delay to wait some time before or after a timestamp that you provide in your workflow payload. Relative delays are great for use cases such as reminders, where you want to wait until a day before a due date to remind your users.
```

I'm finding current language pretty hard to parse. Let's aim to simplify and get this to a tangible use case as quickly as possible.

When configuring a relative delay, you'll specify:

> - The set interval (e.g., 1 day, 2 hours, 30 minutes)

Reviewer: ```suggestion
- The delay interval (e.g., 1 day, 2 hours, 30 minutes)
```
Just to make consistent with bullet that comes after it

> - Whether the delay should occur before or after the dynamic duration

Reviewer 2: ```suggestion
- Whether the delay should occur before or after the dynamic timestamp
```
I wonder if this would be clearer? I know we use "dynamic duration" in the section above, but that's when the delay itself is dynamic based on a dynamically-provided explicit timestamp, where as now we're saying the duration is the interval that is calculated based on that explicit timestamp. I think it's confusing to say both "interval" and "duration" but refer to two separate things

Author: i think you're right was back and forth with this since we call the above step type a dynamic duration, however the more obvious use case for this feature is to use a timestamp

Author: going to keep this comment section unresolved while we get feedback

Author: @cellomatt I removed references to dynamic duration where we aren't explicitly referencing a duration unit format

> - The `key` for the dynamic duration (which can come from your trigger data, recipient data, or other sources)

Reviewer 2: ```suggestion
- The `key` for the dynamic timestamp (which can come from your trigger data, recipient data, or other sources)
```
Same question as above. Or maybe "dynamic duration's timestamp" 🤔

> As in the dynamic duration section above, the key specified must be available to be resolved. If the key is missing or resolves to an invalid value, a corresponding error will be logged on the workflow run, and the delay will be **skipped**.

Reviewer: ```suggestion
As in the dynamic duration section above, the key specified must be available to be resolved. If the key is missing or resolves to an invalid value, a corresponding error will be logged on the workflow run, and the delay will be skipped.
```
don't think you need the extra emphasis here

Relative interval delay functions are helpful for various scenarios, including:

> - Appointment reminders: Send a notification 1 day before an appointment time

Reviewer: ```suggestion
- Appointment reminders: send a notification one day before an appointment time
```

> - Follow-up reminders: send a follow-up message two hours after an event

Author: sounds good - I'll probably remove the ending sentence at the end of the first paragraph. maybe add a bit more explanation instead

> - Follow-up reminders: Send a follow-up message 2 hours after an event

Reviewer: ```suggestion
- Follow-up reminders: send a follow-up message two hours after an event
```
